safekeeper/src/lib.rs

```diff
@@ -95,3 +96,11 @@ impl SafeKeeperConf {
         }
     }
 }
+
+/// Get unique thread id (Rust internal), with ThreadId removed for shorter printing
+pub fn get_tid() -> u64 {
+    let tids = format!("{:?}", thread::current().id());
+    let r = Regex::new(r"ThreadId\((\d+)\)").unwrap();
+    let caps = r.captures(&tids).unwrap();
+    caps.get(1).unwrap().as_str().parse().unwrap()
+}
```

Comment:
https://docs.rs/nix/latest/nix/unistd/fn.getpid.html

Reply:
Yeah, let's remove this shame. We hadn't nix dependency when this was added.